challenge-manager/challenge-tree/paths.go

```diff
@@ -64,12 +64,12 @@ func (ht *RoyalChallengeTree) ComputePathWeight(
 	return pathWeight, nil
 }
 
-type essentialPath []protocol.EdgeId
+type EssentialPath []protocol.EdgeId
 
-type isConfirmableArgs struct {
-	essentialNode         protocol.EdgeId
-	confirmationThreshold uint64
-	blockNum              uint64
+type IsConfirmableArgs struct {
+	EssentialNode         protocol.EdgeId
```

Comment:
Would it be okay to call this `EssentialEdgeId`?

Also, a note for a later PR, we should sweep through this whole repo and replace `Id` with `ID`. https://google.github.io/styleguide/go/decisions#initialisms